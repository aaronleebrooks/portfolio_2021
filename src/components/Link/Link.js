import cx from 'classnames';
import styles from './index.module.css';

function Link(props) {
  return (
    <a className={cx(styles.link, props.className)} href={props.href} target={props.target} rel="noopener noreferrer" download={props.download}>
      {props.children}
    </a>
  );
}

export default Link;
