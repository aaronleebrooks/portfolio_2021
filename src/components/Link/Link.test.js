import { render } from '@testing-library/react';
import Link from './Link';

describe('Link', () => {
  const baseProps = {
    children: "test element"
  };

  test('renders with props', () => {
    const { container } = render(<Link {...baseProps} />);
    expect(container).toMatchSnapshot();
  });
})
