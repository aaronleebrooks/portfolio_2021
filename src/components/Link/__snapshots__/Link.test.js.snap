// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders with props 1`] = `
<div>
  <a
    class="link"
    rel="noopener noreferrer"
  >
    test element
  </a>
</div>
`;
